src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -431,11 +444,29 @@ private HashAlgorithm GetHashAlgorithm(string algorithm)
             {
                 switch (algorithm)
                 {
-                    case "SHA256":
-                    case JwtAlgorithms.RSA_SHA256:
-                    case JwtAlgorithms.ECDSA_SHA256:
-                    case JwtAlgorithms.HMAC_SHA256:
+                    case SecurityAlgorithms.RsaSha1Signature:
```

Comment:
Let's remove support for SHA1, it is not a valid algorithm for JWS.

https://tools.ietf.org/html/rfc7518#section-3